metaseq/scripts/convert_to_singleton.py

```diff
@@ -0,0 +1,133 @@
+
+Particularly useful for converting our models to other repositories.
+
+Usage:
+    $ ls 125m
+    dict.txt
+    gpt2-merges.txt
+    gpt2-vocab.json
+    reshard-model_part-0.pt
+    reshard-model_part-1.pt
+
+    $ python -m metaseq.scripts.convert_to_singleton 125m
+
+    $ ls 125m
+    dict.txt
+    gpt2-merges.txt
+    gpt2-vocab.json
+    reshard-model_part-0.pt
+    reshard-model_part-1.pt
+    restored.pt
+"""
+
+import torch
+import argparse
+import glob
+
+from metaseq import options, tasks, checkpoint_utils, utils
+from metaseq.dataclass.configs import MetaseqConfig
+from metaseq.dataclass.utils import convert_namespace_to_omegaconf
+from metaseq.distributed import utils as dist_utils
+from metaseq.distributed import fsdp_enable_wrap, fsdp_wrap
+from metaseq.distributed.stitch_fsdp_ckpt import glue_megatron_parts
+
+
+def worker_main(cfg: MetaseqConfig):
+    """
+    Load up the model on all workers for Model Parallelism, then
+    unflatten, move to cpu, and save to "restored.pt".
+    """
+    task = tasks.setup_task(cfg.task)
+
+    def _build_model(cfg, task):
+        # hardcoded to cpu & fp16
+        model = task.build_model(cfg.model).half().cuda()
+        return fsdp_wrap(model)
+
+    with fsdp_enable_wrap(
+        cfg.distributed_training,
+        use_sharded_state=cfg.distributed_training.use_sharded_state,
+    ):
+        models, _model_args, _task = checkpoint_utils.load_model_ensemble_and_task(
+            utils.split_paths(cfg.common_eval.path),
+            arg_overrides=None,
+            task=task,
+            suffix=cfg.checkpoint.checkpoint_suffix,
+            strict=True,
+            num_shards=cfg.checkpoint.checkpoint_shard_count,
+            build_model_hook=_build_model,
+        )
+        model = models[0]
+
+    # consolidate everything on rank0
+    mp_size = dist_utils.get_model_parallel_world_size()
+    model_parts = [{} for _ in range(mp_size)]
+    sd = model.module.state_dict()
+    for key in sorted(sd.keys()):
+        tensor = sd[key]
+        gathered = [tensor for _ in range(mp_size)]
+        torch.distributed.all_gather(
+            gathered, tensor, group=dist_utils.get_global_group()
+        )
+        for r, t in enumerate(gathered):
+            model_parts[r][key] = t.cpu()
+
+    glued = glue_megatron_parts(model_parts)
+    # glued['decoder.output_projection.weight'] = glued['decoder.embed_tokens.weight']
+    del glued["decoder.output_projection.weight"]
+
+    if dist_utils.get_global_rank() == 0:
+        with open(cfg.task.data + "/restored.pt", "wb") as f:
+            torch.save(glued, f)
+
+
+def main():
+    # parser to be used like docstring shows
+    real_parser = argparse.ArgumentParser()
+    real_parser.add_argument("location")
+    args = real_parser.parse_args()
+    files = glob.glob(f"{args.location}/reshard*.pt")
+
+    MP = len(files)
+    BPE_MERGES = args.location + "/gpt2-merges.txt"
+    BPE_VOCAB = args.location + "/gpt2-vocab.txt"
```

Comment:
```suggestion
    BPE_VOCAB = args.location + "/gpt2-vocab.json"
```
I believe this needs to be changed?